test-project/src/test_contract_tests.rs

```diff
@@ -0,0 +1,132 @@
+use std::sync::Mutex;
+
+use aleph_client::{pallets::balances::BalanceUserApi, KeyPair, SignedConnection, TxStatus};
+use anyhow::Result;
+use assert2::assert;
+use ink_primitives::AccountId;
+use once_cell::sync::Lazy;
+use rand::RngCore as _;
+use test_contract::{Enum1, Struct1, Struct2};
+
+use crate::test_contract;
+
+static AUTHORITY_MUTEX: Lazy<Mutex<()>> = Lazy::new(|| Mutex::new(()));
+
+async fn connect_and_deploy() -> Result<(SignedConnection, test_contract::Instance)> {
+    let guard = AUTHORITY_MUTEX.lock().unwrap();
+
+    let conn = aleph_client::Connection::new("ws://localhost:9944").await;
+    let alice = aleph_client::keypair_from_string("//Alice");
+    let test_account = random_account();
+
+    aleph_client::SignedConnection::from_connection(conn.clone(), alice)
+        .transfer(
+            test_account.account_id().clone(),
+            alephs(100),
+            TxStatus::InBlock,
+        )
+        .await?;
+
+    drop(guard);
```

Comment:
It's not that. You unlock the mutex by dropping the guard you get from `lock().unwrap()`, but if you don't use it for anything you get warned about an unused variable. And if you `let _ = mutex.lock().unwrap()`, then it gets dropped immediately. Anyway, all this would be solved if the connection itself is in the mutex/lazy.

Reply:
Sure. There's a `Drop` for the `Guard` type so if this was extracted to a method then it would be dropped automatically when it's finished.